jax/interpreters/xla.py

```diff
@@ -1019,6 +1019,7 @@ def f_with_avals(c, avals, xla_args, params):
       wrapped_fun = _tuple_output(wrapped_fun)
     with core.extend_axis_env_nd(zip(axis_env.names, axis_env.sizes)):
       jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(wrapped_fun, avals)
+    jaxpr = core.apply_outfeed_rewriter(jaxpr)
```

Comment:
If this needs to go here, it also needs to go in the other variants within this function, and in the similar function in mlir.py.

But I'm a bit confused by, e.g., what `avals` means here now, because this function changes the jaxpr's signature. How can this be type correct? The caller will not have passed enough arguments to make this work.